--- nfs-utils-2.3.3/support/nfsidmap/Makefile.am.orig	2018-09-06 20:09:08.000000000 +0200
+++ nfs-utils-2.3.3/support/nfsidmap/Makefile.am	2019-01-30 23:56:24.697893513 +0100
@@ -25,6 +25,7 @@
 #  <age> 	The number of previous additional interfaces supported
 #  		by this library.
 
+libnfsidmap_la_CFLAGS = -DPATH_PLUGINS=\"$(pkgplugindir)\"
 libnfsidmap_la_SOURCES = libnfsidmap.c nfsidmap_common.c
 libnfsidmap_la_LDFLAGS = -version-info 1:0:0
 libnfsidmap_la_LIBADD = -ldl ../../support/nfs/libnfsconf.la
