--- nfs-utils-2.4.2/aclocal/kerberos5.m4.orig	2019-11-29 16:24:46.626777231 +0100
+++ nfs-utils-2.4.2/aclocal/kerberos5.m4	2019-11-29 16:27:53.019100791 +0100
@@ -39,8 +39,8 @@
       if test -f $dir/include/gssapi/gssapi_krb5.h -a \
                 \( -f $dir/lib/libgssapi_krb5.a -o \
                    -f $dir/lib/libgssapi_krb5.so -o \
-                   -f $dir/lib32/libgssapi_krb5.a -o \
-                   -f $dir/lib32/libgssapi_krb5.so -o \
+                   -f $dir/libx32/libgssapi_krb5.a -o \
+                   -f $dir/libx32/libgssapi_krb5.so -o \
                    -f $dir/lib64/libgssapi_krb5.a -o \
                    -f $dir/lib64/libgssapi_krb5.so \) ; then
          AC_DEFINE(HAVE_KRB5, 1, [Define this if you have MIT Kerberos libraries])
@@ -53,6 +53,8 @@
       		 -f $dir/include/heimdal/heim_err.h \) -a \
                 \( -f $dir/lib/libroken.a -o \
                    -f $dir/lib64/libroken.a -o \
+                   -f $dir/libx32/libroken.a -o \
+                   -f $dir/libx32/libroken.so -o \
                    -f $dir/lib64/libroken.so -o \
                    -f $dir/lib/libroken.so \) ; then
          K5VERS=`$K5CONFIG --version | head -n 1 | awk '{split($(2),v,"."); if (v@<:@"3"@:>@ == "") v@<:@"3"@:>@ = "0"; print v@<:@"1"@:>@v@<:@"2"@:>@v@<:@"3"@:>@ }'`
--- nfs-utils-2.3.3/utils/nfsdcltrack/sqlite.c.orig	2018-09-06 20:09:08.000000000 +0200
+++ nfs-utils-2.3.3/utils/nfsdcltrack/sqlite.c	2018-12-09 20:41:52.172247084 +0100
@@ -542,8 +542,8 @@
 	int ret;
 	char *err = NULL;
 
-	ret = snprintf(buf, sizeof(buf), "DELETE FROM clients WHERE time < %ld",
-			grace_start);
+	ret = snprintf(buf, sizeof(buf), "DELETE FROM clients WHERE time < %lld",
+			(long long)grace_start);
 	if (ret < 0) {
 		return ret;
 	} else if ((size_t)ret >= sizeof(buf)) {
