--- nfs-utils-1.0.5/configure.in.orig	2003-07-18 01:34:05.000000000 +0200
+++ nfs-utils-1.0.5/configure.in	2003-08-11 20:23:42.584171056 +0200
@@ -84,11 +84,11 @@
 dnl *************************************************************
 dnl * Check for required librarues
 dnl *************************************************************
-AC_CHECK_LIB(socket, main, [LIBSOCKET="-lnsl"])
-AC_CHECK_LIB(nsl, main,	[LIBNSL="-lnsl"])
+AC_CHECK_FUNC(socket, [LIBSOCKET=""], [AC_CHECK_LIB(socket, main, [LIBSOCKET="-lnsl"])])
+AC_CHECK_FUNC(gethostbyname, [LIBNSL=""], [AC_CHECK_LIB(nsl, main, [LIBNSL="-lnsl"])])
 AC_CHECK_LIB(crypt, crypt, [LIBCRYPT="-lcrypt"])
 if test "$knfsd_cv_glibc2" = no; then
-    AC_CHECK_LIB(bsd, daemon, [LIBBSD="-lbsd"])
+    AC_CHECK_FUNC(daemon, , [AC_CHECK_LIB(bsd, daemon, [LIBBSD="-lbsd"])])
 fi
 AC_SUBST(LIBSOCKET)
 AC_SUBST(LIBNSL)
