--- nfs-utils-2.5.4/utils/statd/statd.man.orig	2021-06-10 20:07:47.000000000 +0200
+++ nfs-utils-2.5.4/utils/statd/statd.man	2021-07-04 20:44:39.952652905 +0200
@@ -251,7 +251,7 @@
 If this option is not specified,
 .B rpc.statd
 uses
-.I /var/lib/nfs
+.I /var/lib/nfs/statd
 by default.
 .IP
 After starting,
@@ -431,13 +431,13 @@
 .IR \-\-no\-notify .
 .SH FILES
 .TP 2.5i
-.I /var/lib/nfs/sm
+.I /var/lib/nfs/statd/sm
 directory containing monitor list
 .TP 2.5i
-.I /var/lib/nfs/sm.bak
+.I /var/lib/nfs/statd/sm.bak
 directory containing notify list
 .TP 2.5i
-.I /var/lib/nfs/state
+.I /var/lib/nfs/statd/state
 NSM state number for this host
 .TP 2.5i
 .I /run/run.statd.pid
--- nfs-utils-1.2.2/utils/statd/sm-notify.man~	2010-02-18 13:35:00.000000000 +0100
+++ nfs-utils-1.2.2/utils/statd/sm-notify.man	2010-02-24 17:30:27.917124808 +0100
@@ -186,7 +186,7 @@
 If this option is not specified,
 .B sm-notify
 uses
-.I /var/lib/nfs
+.I /var/lib/nfs/statd
 by default.
 .IP
 After starting,
@@ -287,13 +287,13 @@
 command supports sending notification only via datagram transport protocols.
 .SH FILES
 .TP 2.5i
-.I /var/lib/nfs/sm
+.I /var/lib/nfs/statd/sm
 directory containing monitor list
 .TP 2.5i
-.I /var/lib/nfs/sm.bak
+.I /var/lib/nfs/statd/sm.bak
 directory containing notify list
 .TP 2.5i
-.I /var/lib/nfs/state
+.I /var/lib/nfs/statd/state
 NSM state number for this host
 .TP 2.5i
 .I /proc/sys/fs/nfs/nsm_local_state
