--- nfs-utils-1.1.5/utils/mount/mount.c~	2009-03-05 12:42:56.000000000 +0100
+++ nfs-utils-1.1.5/utils/mount/mount.c	2009-05-20 16:44:08.581980119 +0200
@@ -94,6 +94,7 @@
   { "sync",     0, 0, MS_SYNCHRONOUS},  /* synchronous I/O */
   { "async",    0, 1, MS_SYNCHRONOUS},  /* asynchronous I/O */
   { "dirsync",  0, 0, MS_DIRSYNC},      /* synchronous directory modifications */
+  { "union",	0, 0, MS_UNION  },	/* Union mount */
   { "remount",  0, 0, MS_REMOUNT},      /* Alter flags of mounted FS */
   { "bind",     0, 0, MS_BIND   },      /* Remount part of tree elsewhere */
   { "rbind",    0, 0, MS_BIND|MS_REC }, /* Idem, plus mounted subtrees */
--- nfs-utils-1.1.5/utils/mount/mount_constants.h~	2009-03-05 12:42:56.000000000 +0100
+++ nfs-utils-1.1.5/utils/mount/mount_constants.h	2009-05-20 16:42:50.292974566 +0200
@@ -21,6 +21,9 @@
 if we have a stack or plain mount - mount atop of it, forming a stack. */
 #define	MS_OVER		0x200	/* 512 */
 #endif
+#ifndef MS_UNION
+#define MS_UNION	0x100	/* 256: Mount on top of a union */
+#endif
 #ifndef MS_NOATIME
 #define MS_NOATIME	0x400	/* 1024: Do not update access times. */
 #endif
