--- nfs-utils-2.5.3/aclocal/kerberos5.m4.orig	2021-02-24 21:11:25.667510884 +0100
+++ nfs-utils-2.5.3/aclocal/kerberos5.m4	2021-02-24 22:08:08.296189188 +0100
@@ -39,8 +39,8 @@
       if test -f $dir/include/gssapi/gssapi_krb5.h -a \
                 \( -f $dir/lib/libgssapi_krb5.a -o \
                    -f $dir/lib/libgssapi_krb5.so -o \
-                   -f $dir/lib32/libgssapi_krb5.a -o \
-                   -f $dir/lib32/libgssapi_krb5.so -o \
+                   -f $dir/libx32/libgssapi_krb5.a -o \
+                   -f $dir/libx32/libgssapi_krb5.so -o \
                    -f $dir/lib64/libgssapi_krb5.a -o \
                    -f $dir/lib64/libgssapi_krb5.so -o \
                    -f $dir/lib/$(uname -m)-linux-gnu/libgssapi_krb5.a -o \
@@ -55,6 +55,8 @@
       		 -f $dir/include/heimdal/heim_err.h \) -a \
                 \( -f $dir/lib/libroken.a -o \
                    -f $dir/lib64/libroken.a -o \
+                   -f $dir/libx32/libroken.a -o \
+                   -f $dir/libx32/libroken.so -o \
                    -f $dir/lib64/libroken.so -o \
                    -f $dir/lib/libroken.so \) ; then
          K5VERS=`$K5CONFIG --version | head -n 1 | awk '{split($(2),v,"."); if (v@<:@"3"@:>@ == "") v@<:@"3"@:>@ = "0"; print v@<:@"1"@:>@v@<:@"2"@:>@v@<:@"3"@:>@ }'`
