diff -urN nfs-utils-1.0.7.org/configure.in nfs-utils-1.0.7/configure.in
--- nfs-utils-1.0.7.org/configure.in	2004-12-17 05:09:09.000000000 +0100
+++ nfs-utils-1.0.7/configure.in	2005-02-27 21:11:35.317266320 +0100
@@ -161,7 +161,7 @@
     dnl MIT Kerberos on Fedora Core 1
     K5CONFIG=""
     if test -f $dir/bin/krb5-config; then
-      K5CONFIG=$dir/bin/krb5-config
+      K5CONFIG="$dir/bin/krb5-config"
     elif test -f "/usr/kerberos/bin/krb5-config"; then
       K5CONFIG="/usr/kerberos/bin/krb5-config"
     fi
@@ -188,7 +188,7 @@
       dnl of Heimdal Kerberos on SuSe
       elif test \( -f $dir/include/heim_err.h -o\
       		 -f $dir/include/heimdal/heim_err.h \) -a \
-                -f $dir/lib/libroken.a; then
+                -f $dir/lib/libroken.so; then
          AC_DEFINE(HAVE_HEIMDAL)
          KRBDIR="$dir"
          K5VERS=`$K5CONFIG --version | head -1 | awk '{split($2,v,"."); print v[[1]]v[[2]]v[[3]] }'`
@@ -200,9 +200,10 @@
            AC_DEFINE(HAVE_SET_ALLOWABLE_ENCTYPES),,$KRBLIB)
          AC_CHECK_LIB(gssapi, gss_krb5_ccache_name,
            AC_DEFINE(HAVE_GSS_KRB5_CCACHE_NAME),,$KRBLIB)
+	 CFLAGS="$CFLAGS -I/usr/include/et"
         break
       fi
-      CFLAGS=$CFLAGS `K5CONFIG --cflags`
+      CFLAGS="$CFLAGS -I/usr/include/et `$K5CONFIG --cflags`"
     fi
   done
   dnl We didn't find a usable Kerberos environment
diff -urN nfs-utils-1.0.7.org/support/lib/Makefile nfs-utils-1.0.7/support/lib/Makefile
--- nfs-utils-1.0.7.org/support/lib/Makefile	2004-10-19 02:22:58.000000000 +0200
+++ nfs-utils-1.0.7/support/lib/Makefile	2005-02-27 22:21:21.819821192 +0100
@@ -1,7 +1,7 @@
 
 include	$(TOP)rules.mk
 
-LIBS	= libnfs.a libexport.a libmisc.a librpc.a libgssapi.a
+LIBS	= libnfs.a libexport.a libmisc.a librpc.a
 
 all install::	$(LIBS)
 	@:
diff -urN nfs-utils-1.0.7.org/utils/gssd/krb5_util.c nfs-utils-1.0.7/utils/gssd/krb5_util.c
--- nfs-utils-1.0.7.org/utils/gssd/krb5_util.c	2004-10-19 02:23:08.000000000 +0200
+++ nfs-utils-1.0.7/utils/gssd/krb5_util.c	2005-02-27 22:30:21.687748792 +0100
@@ -120,6 +120,8 @@
 #include "gss_oids.h"
 #include "krb5_util.h"
 
+#define krb5_free_unparsed_name(con,val) krb5_xfree((char *)(val))
+
 /* Global list of principals/cache file names for machine credentials */
 struct gssd_k5_kt_princ *gssd_k5_kt_princ_list = NULL;
 
@@ -466,8 +468,8 @@
 		}
 		printerr(2, "Processing keytab entry for principal '%s'\n",
 			 pname);
-		if ( (kte.principal->data[0].length == GSSD_SERVICE_NAME_LEN) &&
-		     (strncmp(kte.principal->data[0].data, GSSD_SERVICE_NAME,
+		if ( (kte.principal->name.name_string.len == GSSD_SERVICE_NAME_LEN) &&
+		     (strncmp(kte.principal->name.name_string.val[0], GSSD_SERVICE_NAME,
 			      GSSD_SERVICE_NAME_LEN) == 0) &&
 		     (!gssd_have_realm_ple(&kte.principal->realm)) ) {
 			printerr(2, "We will use this entry (%s)\n", pname);
@@ -484,8 +486,8 @@
 			ple->ccname = NULL;
 			ple->endtime = 0;
 			if ((ple->realm =
-				strndup(kte.principal->realm.data,
-					kte.principal->realm.length))
+				strndup(kte.principal->realm,
+					strlen(kte.principal->realm)))
 					== NULL) {
 				printerr(0, "ERROR: %s while copying realm to "
 					    "principal list entry\n",
diff -urN nfs-utils-1.0.7.org/utils/gssd/Makefile nfs-utils-1.0.7/utils/gssd/Makefile
--- nfs-utils-1.0.7.org/utils/gssd/Makefile	2004-10-19 02:23:08.000000000 +0200
+++ nfs-utils-1.0.7/utils/gssd/Makefile	2005-02-27 22:22:02.385654248 +0100
@@ -6,7 +6,7 @@
 PREFIX	= rpc.
 OBJS	= gssd.o gssd_main_loop.o gssd_proc.o err_util.o gss_util.o \
 	  gss_oids.o context.o context_heimdal.o krb5_util.o
-LIBDEPS	= $(TOP)support/lib/librpc.a $(TOP)support/lib/libgssapi.a
+LIBDEPS	= $(TOP)support/lib/librpc.a
 LIBS	= -Wl,-rpath=$(KRBDIR)/lib -lrpc -lgssapi -ldl $(KRBLIB)
 MAN8	= gssd
 
diff -urN nfs-utils-1.0.7.org/utils/svcgssd/Makefile nfs-utils-1.0.7/utils/svcgssd/Makefile
--- nfs-utils-1.0.7.org/utils/svcgssd/Makefile	2004-11-22 06:41:13.000000000 +0100
+++ nfs-utils-1.0.7/utils/svcgssd/Makefile	2005-02-27 22:22:26.624969312 +0100
@@ -6,7 +6,7 @@
 PREFIX	= rpc.
 OBJS	= svcgssd.o svcgssd_main_loop.o svcgssd_proc.o err_util.o gss_util.o \
 	  gss_oids.o context.o context_heimdal.o cacheio.o svcgssd_mech2file.o
-LIBDEPS	= $(TOP)support/lib/librpc.a $(TOP)support/lib/libgssapi.a
+LIBDEPS	= $(TOP)support/lib/librpc.a
 LIBS	= -Wl,-rpath=$(KRBDIR)/lib -lrpc -lgssapi -ldl $(KRBLIB) -lnfsidmap
 MAN8	= svcgssd
 
