diff -u -p -r1.2 lockd.c
--- utils/lockd/lockd.c	2001/01/25 21:11:29	1.2
+++ utils/lockd/lockd.c	2001/05/28 15:32:21
@@ -9,6 +9,8 @@
 #include "config.h"
 
 #include <stdio.h>
+#include <unistd.h>
+#include <string.h>
 #include <errno.h>
 #include "nfslib.h"
 
@@ -21,6 +23,12 @@ main(int argc, char **argv)
 
 	if (argc > 1)
 		usage (argv [0]);
+
+	if (chdir(NFS_STATEDIR)) {
+		fprintf(stderr, "%s: chdir(%s) failed: %s\n",
+			argv [0], NFS_STATEDIR, strerror(errno));
+		exit(1);
+	}
 
 	if ((error = lockdsvc()) < 0) {
 		if (errno == EINVAL)
diff -u -p -r1.8 mountd.c
--- utils/mountd/mountd.c	2001/02/11 01:12:02	1.8
+++ utils/mountd/mountd.c	2001/05/28 15:32:22
@@ -14,6 +14,7 @@
 #include <arpa/inet.h>
 #include <unistd.h>
 #include <stdlib.h>
+#include <string.h>
 #include <getopt.h>
 #include <errno.h>
 #include <fcntl.h>
@@ -439,6 +440,12 @@ main(int argc, char **argv)
 	/* No more arguments allowed. */
 	if (optind != argc || !(nfs_version & 0x7))
 		usage(argv [0], 1);
+
+	if (chdir(NFS_STATEDIR)) {
+		fprintf(stderr, "%s: chdir(%s) failed: %s\n",
+			argv [0], NFS_STATEDIR, strerror(errno));
+		exit(1);
+	}
 
 	/* Initialize logging. */
 /*	xlog_open("mountd"); */
diff -u -p -r1.1.1.1 nfsd.c
--- utils/nfsd/nfsd.c	1999/10/18 23:21:12	1.1.1.1
+++ utils/nfsd/nfsd.c	2001/05/28 15:32:22
@@ -11,6 +11,9 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
 #include <getopt.h>
 #include "nfslib.h"
 
@@ -41,6 +44,12 @@ main(int argc, char **argv)
 		default:
 			usage(argv[0]);
 		}
+	}
+
+	if (chdir(NFS_STATEDIR)) {
+		fprintf(stderr, "%s: chdir(%s) failed: %s\n",
+			argv [0], NFS_STATEDIR, strerror(errno));
+		exit(1);
 	}
 
 	if (optind < argc) {
