--- nfs-utils-2.4.3/utils/nfsdcltrack/nfsdcltrack.c.orig	2020-05-17 08:13:59.744382256 +0200
+++ nfs-utils-2.4.3/utils/nfsdcltrack/nfsdcltrack.c	2020-05-17 08:19:47.068916900 +0200
@@ -525,7 +525,7 @@
 	if (*tail)
 		return -EINVAL;
 
-	xlog(D_GENERAL, "%s: grace done. gracetime=%ld", __func__, gracetime);
+	xlog(D_GENERAL, "%s: grace done. gracetime=%lld", __func__, (long long)gracetime);
 
 	ret = sqlite_remove_unreclaimed(gracetime);
 
