diff -uNr nfs-utils-0.2.1/support/nfs/rpcmisc.c nfs-utils-0.2.1-eepro100/support/nfs/rpcmisc.c
--- nfs-utils-0.2.1/support/nfs/rpcmisc.c	Sun Apr 30 00:14:56 2000
+++ nfs-utils-0.2.1-eepro100/support/nfs/rpcmisc.c	Fri Jan 19 13:51:45 2001
@@ -176,7 +176,8 @@
 		int sblen, rblen;
 
 		/* 1024 for rpc & transport overheads */
-		sblen = rblen = socksz + 1024;
+		/* but, 2048 needed for eepro100 */
+		sblen = rblen = socksz + 2048;
 		if (setsockopt(s, SOL_SOCKET, SO_SNDBUF, &sblen, sizeof sblen) < 0 ||
 		    setsockopt(s, SOL_SOCKET, SO_RCVBUF, &rblen, sizeof rblen) < 0)
 			xlog(L_ERROR, "setsockopt failed: %s\n", strerror(errno));
