diff -urN nfs-utils-1.0.7.org/utils/gssd/context_heimdal.c nfs-utils-1.0.7/utils/gssd/context_heimdal.c
--- nfs-utils-1.0.7.org/utils/gssd/context_heimdal.c	2005-07-10 12:14:01.526867000 +0200
+++ nfs-utils-1.0.7/utils/gssd/context_heimdal.c	2005-07-10 12:14:32.815110736 +0200
@@ -46,6 +46,20 @@
 #include "gss_oids.h"
 #include "write_bytes.h"
 
+typedef struct gss_ctx_id_t_desc_struct {
+  struct krb5_auth_context_data *auth_context;
+  gss_name_t source, target;
+  OM_uint32 flags;
+  enum { LOCAL = 1, OPEN = 2,
+         COMPAT_OLD_DES3 = 4,
+         COMPAT_OLD_DES3_SELECTED = 8,
+  } more_flags;
+  struct krb5_ticket *ticket;
+  OM_uint32 lifetime;
+  pthread_mutex_t ctx_id_mutex;
+  struct gss_msg_order *order;
+} gss_ctx_id_t_desc;
+
 #define MAX_CTX_LEN 4096
 
 int write_heimdal_keyblock(char **p, char *end, krb5_keyblock *key)
