--- ./support/nfs/xlog.c.orig	Fri Jun 20 19:29:41 2003
+++ ./support/nfs/xlog.c	Fri Jun 20 19:31:15 2003
@@ -142,9 +142,9 @@
 		return;
 
 	va_start(args, fmt);
-	vsnprintf(buff, sizeof (buff), fmt, args);
+	vsnprintf(buff, sizeof (buff) - 2, fmt, args);
 	va_end(args);
-	buff[sizeof (buff) - 1] = 0;
+	buff[sizeof (buff) - 2] = 0;
 
 	if ((n = strlen(buff)) > 0 && buff[n-1] != '\n') {
 		buff[n++] = '\n'; buff[n++] = '\0';
