--- nfs-utils-1.1.0-rc1/utils/mount/mount.c~	2007-03-29 05:55:33.000000000 +0200
+++ nfs-utils-1.1.0-rc1/utils/mount/mount.c	2007-04-05 18:58:38.000000000 +0200
@@ -508,30 +508,30 @@
 	    }
 	}
 
-	if (nfs_mount_vers == 4)
-		mnt_err = nfs4mount(spec, mount_point, &flags, &extra_opts, &mount_opts, 0);
-	else {
-		int need_statd = 0;
-		mnt_err = nfsmount(spec, mount_point, &flags,
-				   &extra_opts, &mount_opts,
-				   0, &need_statd);
-		if (!mnt_err && !fake && need_statd) {
-			if (!start_statd()) {
-				fprintf(stderr,
-					"%s: rpc.statd is not running but is "
-					"required for remote locking\n"
-					"   Either use \"-o nolocks\" to keep "
-					"locks local, or start statd.\n",
-					progname);
-				exit(1);
+	if (!fake) {
+		if (nfs_mount_vers == 4)
+			mnt_err = nfs4mount(spec, mount_point, &flags, &extra_opts, &mount_opts, 0);
+		else {
+			int need_statd = 0;
+			mnt_err = nfsmount(spec, mount_point, &flags,
+					   &extra_opts, &mount_opts,
+					   0, &need_statd);
+			if (!mnt_err && need_statd) {
+				if (!start_statd()) {
+					fprintf(stderr,
+						"%s: rpc.statd is not running but is "
+						"required for remote locking\n"
+						"   Either use \"-o nolocks\" to keep "
+						"locks local, or start statd.\n",
+						progname);
+					exit(1);
+				}
 			}
 		}
-	}
 
-	if (mnt_err)
-		exit(EX_FAIL);
+		if (mnt_err)
+			exit(EX_FAIL);
 
-	if (!fake) {
 		mnt_err = do_mount_syscall(spec, mount_point,
 					   nfs_mount_vers == 4 ? "nfs4" : "nfs",
 					   flags & ~(MS_USER|MS_USERS) ,
