--- nfs-utils-1.0.11/utils/gssd/context_heimdal.c.orig	2007-02-25 20:42:02.000000000 +0100
+++ nfs-utils-1.0.11/utils/gssd/context_heimdal.c	2007-03-10 22:48:54.599236929 +0100
@@ -46,6 +46,22 @@
 #include "err_util.h"
 #include "gss_oids.h"
 #include "write_bytes.h"
+#include "context.h"
+
+typedef struct gss_ctx_id_t_desc_struct {
+  struct krb5_auth_context_data *auth_context;
+  gss_name_t source, target;
+  OM_uint32 flags;
+  enum { LOCAL = 1, OPEN = 2,
+         COMPAT_OLD_DES3 = 4,
+         COMPAT_OLD_DES3_SELECTED = 8,
+  } more_flags;
+  struct krb5_ticket *ticket;
+  OM_uint32 lifetime;
+  pthread_mutex_t ctx_id_mutex;
+  struct gss_msg_order *order;
+} gss_ctx_id_t_desc;
+
 
 int write_heimdal_keyblock(char **p, char *end, krb5_keyblock *key)
 {
