--- nfs-utils-2.5.3/Makefile.am.orig	2021-02-24 20:53:15.487950347 +0100
+++ nfs-utils-2.5.3/Makefile.am	2021-02-24 20:58:24.117204744 +0100
@@ -27,8 +27,6 @@
 	touch $(DESTDIR)$(statedir)/rmtab; chmod 644 $(DESTDIR)$(statedir)/rmtab
 	mkdir -p $(DESTDIR)$(statdpath)/sm $(DESTDIR)$(statdpath)/sm.bak
 	touch $(DESTDIR)$(statdpath)/state
-	chmod go-rwx $(DESTDIR)$(statdpath)/sm $(DESTDIR)$(statdpath)/sm.bak $(DESTDIR)$(statdpath)/state
-	-chown $(statduser) $(DESTDIR)$(statdpath)/sm $(DESTDIR)$(statdpath)/sm.bak $(DESTDIR)$(statdpath)/state
 
 uninstall-hook:
 	rm -f $(DESTDIR)$(statedir)/xtab
