--- nfs-utils-1.0.12/utils/mount/mount.c~	2007-03-22 14:43:13.000000000 +0100
+++ nfs-utils-1.0.12/utils/mount/mount.c	2007-03-22 14:45:06.000000000 +0100
@@ -492,30 +492,31 @@
 		}
 	}
 
-	if (nfs_mount_vers == 4) {
-		mnt_err = nfs4mount(spec, mount_point, &flags, 
-			&extra_opts, &mount_opts, 0);
-	}
-	else {
-		if (!strcmp(progname, "mount.nfs")) {
-			mnt_err = nfsmount(spec, mount_point, &flags,
-					&extra_opts, &mount_opts,  0);
+	if (!fake) {
+		if (nfs_mount_vers == 4) {
+			mnt_err = nfs4mount(spec, mount_point, &flags, 
+				&extra_opts, &mount_opts, 0);
+		}
+		else {
+			if (!strcmp(progname, "mount.nfs")) {
+				mnt_err = nfsmount(spec, mount_point, &flags,
+						&extra_opts, &mount_opts,  0);
+			}
+		}
+
+		if (mnt_err)
+			exit(EX_FAIL);
+
+		mnt_err = do_mount_syscall(spec, mount_point,
+				nfs_mount_vers == 4 ? "nfs4" : "nfs", flags, mount_opts);
+
+		if(mnt_err) {
+			mount_error(mount_point);
+			exit(EX_FAIL);
 		}
-	}
-	if (fake)
-		return 0;
-	if (mnt_err)
-		exit(EX_FAIL);
-
-	mnt_err = do_mount_syscall(spec, mount_point,
-			nfs_mount_vers == 4 ? "nfs4" : "nfs", flags, mount_opts);
-
-	if(mnt_err) {
-		mount_error(mount_point);
-		exit(EX_FAIL);
 	}
 
-	if(!nomtab) {
+	if (!nomtab) {
 		add_mtab(spec, mount_point, nfs_mount_vers == 4 ? "nfs4" : "nfs",
 			 flags, extra_opts, 0, 0);
 	}
