--- nfs-utils-1.0.7/utils/rquotad/rquota_server.c.orig	2005-08-02 10:02:31.000000000 -0400
+++ nfs-utils-1.0.7/utils/rquotad/rquota_server.c	2005-08-02 10:08:37.000000000 -0400
@@ -173,7 +173,11 @@ getquota_rslt *getquotainfo(int flags, c
          rquota = &result.getquota_rslt_u.gqr_rquota;
          rquota->rq_bhardlimit = dq_dqb.dqb_bhardlimit;
          rquota->rq_bsoftlimit = dq_dqb.dqb_bsoftlimit;;
+#if _LINUX_QUOTA_VERSION < 2
          rquota->rq_curblocks = dq_dqb.dqb_curblocks;
+#else
+         rquota->rq_curblocks = dq_dqb.dqb_curspace;
+#endif
          rquota->rq_fhardlimit = dq_dqb.dqb_ihardlimit;
          rquota->rq_fsoftlimit = dq_dqb.dqb_isoftlimit;
          rquota->rq_curfiles = dq_dqb.dqb_curinodes;


