--- nfs-utils-1.0.9/configure.in.orig	2006-07-17 12:23:46.164440000 +0200
+++ nfs-utils-1.0.9/configure.in	2006-07-18 13:06:53.474440000 +0200
@@ -164,7 +164,8 @@
        AC_MSG_ERROR([variable PKG_CONFIG_PATH to /usr/local/lib/pkgconfig])
       ]
      )
-    PKG_CHECK_MODULES(GSSAPI, libgssapi >= 0.9)
+    AC_CHECK_LIB(gssapi, main, [GSSAPI_LIBS="-lgssapi"], [AC_MSG_ERROR([Unable to locate libgssapi])])
+    AC_SUBST(GSSAPI_LIBS)
     fi
 
 fi
diff -ur nfs-utils-1.0.8/utils/gssd/context.c nfs-utils-1.0.8.heimdal/utils/gssd/context.c
--- nfs-utils-1.0.8/utils/gssd/context.c	2006-03-28 02:48:27.000000000 +0200
+++ nfs-utils-1.0.8.heimdal/utils/gssd/context.c	2006-05-05 20:03:21.000000000 +0200
@@ -32,7 +32,7 @@
 #include <stdio.h>
 #include <syslog.h>
 #include <string.h>
-#include <gssapi/gssapi.h>
+#include <gssapi.h>
 #include <rpc/rpc.h>
 #include <rpc/auth_gss.h>
 #include "gss_util.h"
diff -ur nfs-utils-1.0.8/utils/gssd/context_mit.c nfs-utils-1.0.8.heimdal/utils/gssd/context_mit.c
--- nfs-utils-1.0.8/utils/gssd/context_mit.c	2006-04-10 01:51:59.000000000 +0200
+++ nfs-utils-1.0.8.heimdal/utils/gssd/context_mit.c	2006-05-05 20:05:28.000000000 +0200
@@ -32,7 +32,7 @@
 #include <stdio.h>
 #include <syslog.h>
 #include <string.h>
-#include <gssapi/gssapi.h>
+#include <gssapi.h>
 #include <rpc/rpc.h>
 #include <rpc/auth_gss.h>
 #include "gss_util.h"
diff -ur nfs-utils-1.0.8/utils/gssd/context_spkm3.c nfs-utils-1.0.8.heimdal/utils/gssd/context_spkm3.c
--- nfs-utils-1.0.8/utils/gssd/context_spkm3.c	2006-04-10 01:51:59.000000000 +0200
+++ nfs-utils-1.0.8.heimdal/utils/gssd/context_spkm3.c	2006-05-05 20:05:36.000000000 +0200
@@ -32,7 +32,7 @@
 #include <stdio.h>
 #include <syslog.h>
 #include <string.h>
-#include <gssapi/gssapi.h>
+#include <gssapi.h>
 #include <rpc/rpc.h>
 #include <rpc/auth_gss.h>
 #include "gss_util.h"
diff -ur nfs-utils-1.0.8/utils/gssd/gss_oids.c nfs-utils-1.0.8.heimdal/utils/gssd/gss_oids.c
--- nfs-utils-1.0.8/utils/gssd/gss_oids.c	2005-08-26 03:20:13.000000000 +0200
+++ nfs-utils-1.0.8.heimdal/utils/gssd/gss_oids.c	2006-05-05 20:03:44.000000000 +0200
@@ -29,7 +29,7 @@
 */
 
 #include <sys/types.h>
-#include <gssapi/gssapi.h>
+#include <gssapi.h>
 
 /* from kerberos source, gssapi_krb5.c */
 gss_OID_desc krb5oid =
diff -ur nfs-utils-1.0.8/utils/gssd/gss_util.c nfs-utils-1.0.8.heimdal/utils/gssd/gss_util.c
--- nfs-utils-1.0.8/utils/gssd/gss_util.c	2005-08-26 03:27:18.000000000 +0200
+++ nfs-utils-1.0.8.heimdal/utils/gssd/gss_util.c	2006-05-05 20:04:08.000000000 +0200
@@ -72,7 +72,7 @@
 #include <sys/param.h>
 #include <netdb.h>
 #include <fcntl.h>
-#include <gssapi/gssapi.h>
+#include <gssapi.h>
 #if defined(HAVE_KRB5) && !defined(GSS_C_NT_HOSTBASED_SERVICE)
 #include <gssapi/gssapi_generic.h>
 #define GSS_C_NT_HOSTBASED_SERVICE gss_nt_service_name
diff -ur nfs-utils-1.0.8/utils/gssd/gssd.h nfs-utils-1.0.8.heimdal/utils/gssd/gssd.h
--- nfs-utils-1.0.8/utils/gssd/gssd.h	2006-03-28 02:50:03.000000000 +0200
+++ nfs-utils-1.0.8.heimdal/utils/gssd/gssd.h	2006-05-05 20:04:27.000000000 +0200
@@ -33,7 +33,7 @@
 
 #include <sys/types.h>
 #include <sys/queue.h>
-#include <gssapi/gssapi.h>
+#include <gssapi.h>
 
 #define MAX_FILE_NAMELEN	32
 #define FD_ALLOC_BLOCK		32
diff -ur nfs-utils-1.0.8/utils/gssd/gssd_proc.c nfs-utils-1.0.8.heimdal/utils/gssd/gssd_proc.c
--- nfs-utils-1.0.8/utils/gssd/gssd_proc.c	2006-04-10 11:57:20.000000000 +0200
+++ nfs-utils-1.0.8.heimdal/utils/gssd/gssd_proc.c	2006-05-05 20:04:35.000000000 +0200
@@ -59,7 +59,7 @@
 #include <signal.h>
 #include <unistd.h>
 #include <errno.h>
-#include <gssapi/gssapi.h>
+#include <gssapi.h>
 #include <netdb.h>
 
 #include "gssd.h"
diff -ur nfs-utils-1.0.8/utils/gssd/krb5_util.c nfs-utils-1.0.8.heimdal/utils/gssd/krb5_util.c
--- nfs-utils-1.0.8/utils/gssd/krb5_util.c	2006-03-28 02:50:03.000000000 +0200
+++ nfs-utils-1.0.8.heimdal/utils/gssd/krb5_util.c	2006-05-05 20:04:53.000000000 +0200
@@ -107,7 +107,7 @@
 #include <dirent.h>
 #include <errno.h>
 #include <time.h>
-#include <gssapi/gssapi.h>
+#include <gssapi.h>
 #ifdef USE_PRIVATE_KRB5_FUNCTIONS
 #include <gssapi/gssapi_krb5.h>
 #endif
diff -ur nfs-utils-1.0.8/utils/gssd/svcgssd.h nfs-utils-1.0.8.heimdal/utils/gssd/svcgssd.h
--- nfs-utils-1.0.8/utils/gssd/svcgssd.h	2006-03-27 02:30:09.000000000 +0200
+++ nfs-utils-1.0.8.heimdal/utils/gssd/svcgssd.h	2006-05-05 20:05:02.000000000 +0200
@@ -33,7 +33,7 @@
 
 #include <sys/types.h>
 #include <sys/queue.h>
-#include <gssapi/gssapi.h>
+#include <gssapi.h>
 
 void handle_nullreq(FILE *f);
 void gssd_run(void);
--- nfs-utils-1.0.11/utils/gssd/svcgssd_mech2file.c.orig	2007-02-25 20:42:02.000000000 +0100
+++ nfs-utils-1.0.11/utils/gssd/svcgssd_mech2file.c	2007-03-10 23:57:23.421385359 +0100
@@ -36,7 +36,7 @@
 
 #include "config.h"
 
-#include <gssapi/gssapi.h>
+#include <gssapi.h>
 #include <string.h>
 
 
--- nfs-utils-1.0.12/utils/gssd/gssd.h~	2007-03-20 19:16:11.000000000 +0100
+++ nfs-utils-1.0.12/utils/gssd/gssd.h	2007-03-20 19:28:43.000000000 +0100
@@ -47,7 +47,7 @@
 #define GSSD_DEFAULT_CRED_DIR			"/tmp"
 #define GSSD_DEFAULT_CRED_PREFIX		"krb5cc_"
 #define GSSD_DEFAULT_MACHINE_CRED_SUFFIX	"machine"
-#define GSSD_DEFAULT_KEYTAB_FILE		"/etc/krb5.keytab"
+#define GSSD_DEFAULT_KEYTAB_FILE		"/etc/heimdal/krb5.keytab"
 #define GSSD_SERVICE_NAME			"nfs"
 #define GSSD_SERVICE_NAME_LEN			3
 
--- nfs-utils-1.0.12/utils/gssd/gssd.man~	2007-02-27 05:55:40.000000000 +0100
+++ nfs-utils-1.0.12/utils/gssd/gssd.man	2007-03-20 19:29:18.000000000 +0100
@@ -31,7 +31,7 @@
 to use the keys for principals nfs/hostname in
 .I keytab
 to obtain machine credentials.
-The default value is "/etc/krb5.keytab".
+The default value is "/etc/heimdal/krb5.keytab".
 .\".TP
 .\".B -m
 .\"Ordinarily,
--- nfs-utils-1.0.12/utils/gssd/svcgssd.c~	2007-02-27 05:55:40.000000000 +0100
+++ nfs-utils-1.0.12/utils/gssd/svcgssd.c	2007-03-20 19:29:27.000000000 +0100
@@ -232,7 +232,7 @@
                 printerr(0, "unable to obtain root (machine) credentials\n");
                 printerr(0, "do you have a keytab entry for "
 			    "nfs/<your.host>@<YOUR.REALM> in "
-			    "/etc/krb5.keytab?\n");
+			    "/etc/heimdal/krb5.keytab?\n");
 		exit(1);
 	}
 
