diff -urN knfsd-1.4.6.orig/config.mk.in knfsd-1.4.6/config.mk.in
--- knfsd-1.4.6.orig/config.mk.in	Sat Jul 31 22:28:53 1999
+++ knfsd-1.4.6/config.mk.in	Sat Jul 31 22:29:10 1999
@@ -4,9 +4,9 @@
 
 prefix		= $(install_prefix)@prefix@
 exec_prefix	= $(install_prefix)@prefix@
-BINDIR		= @bindir@
-SBINDIR		= @sbindir@
-MANDIR		= @mandir@
+BINDIR		= $(install_prefix)@bindir@
+SBINDIR		= $(install_prefix)@sbindir@
+MANDIR		= $(install_prefix)@mandir@
 STATEDIR	= $(install_prefix)@statedir@
 
 ##################################################################
