--- nfs-utils-1.0.12/utils/mount/nfs4mount.c~	2007-03-31 22:46:53.438974675 +0200
+++ nfs-utils-1.0.12/utils/mount/nfs4mount.c	2007-03-31 22:52:34.162391396 +0200
@@ -52,7 +52,7 @@
 extern int verbose;
 extern int sloppy;
 
-char *IDMAPLCK = DEFAULT_DIR "/rpcidmapd";
+char *IDMAPLCK = DEFAULT_DIR "/idmapd";
 #define idmapd_check() do { \
 	if (access(IDMAPLCK, F_OK)) { \
 		printf(_("Warning: rpc.idmapd appears not to be running.\n" \
@@ -60,7 +60,7 @@
 	} \
 } while(0);
 
-char *GSSDLCK = DEFAULT_DIR "/rpcgssd";
+char *GSSDLCK = DEFAULT_DIR "/gssd";
 #define gssd_check() do { \
 		if (access(GSSDLCK, F_OK)) { \
 			printf(_("Warning: rpc.gssd appears not to be running.\n")); \
