--- nfs-utils-1.0.12/utils/gssd/gssd.h~	2007-03-31 01:30:29.173040651 +0200
+++ nfs-utils-1.0.12/utils/gssd/gssd.h	2007-03-31 01:42:42.994858774 +0200
@@ -47,7 +47,7 @@
 #define GSSD_DEFAULT_CRED_DIR			"/tmp"
 #define GSSD_DEFAULT_CRED_PREFIX		"krb5cc_"
 #define GSSD_DEFAULT_MACHINE_CRED_SUFFIX	"machine"
-#define GSSD_DEFAULT_KEYTAB_FILE		"/etc/krb5.keytab"
+#define GSSD_DEFAULT_KEYTAB_FILE		"/var/lib/kerberos/krb5.keytab"
 #define GSSD_SERVICE_NAME			"nfs"
 #define GSSD_SERVICE_NAME_LEN			3
 
--- nfs-utils-1.0.12/utils/gssd/gssd.man~	2007-02-27 05:55:40.000000000 +0100
+++ nfs-utils-1.0.12/utils/gssd/gssd.man	2007-03-31 01:43:00.543858834 +0200
@@ -31,7 +31,7 @@
 to use the keys for principals nfs/hostname in
 .I keytab
 to obtain machine credentials.
-The default value is "/etc/krb5.keytab".
+The default value is "/var/lib/kerberos/krb5.keytab".
 .\".TP
 .\".B -m
 .\"Ordinarily,
--- nfs-utils-1.0.12/utils/gssd/svcgssd.c~	2007-02-27 05:55:40.000000000 +0100
+++ nfs-utils-1.0.12/utils/gssd/svcgssd.c	2007-03-31 01:44:28.936896066 +0200
@@ -232,7 +232,7 @@
                 printerr(0, "unable to obtain root (machine) credentials\n");
                 printerr(0, "do you have a keytab entry for "
 			    "nfs/<your.host>@<YOUR.REALM> in "
-			    "/etc/krb5.keytab?\n");
+			    "/var/lib/kerberos/krb5.keytab?\n");
 		exit(1);
 	}
 
